File: Homework/HW2/Assignment-2_Final/Matlab/Hm2_P6b.m
%%%
%Statistical Computing for Scientists and Engineers
%Homework 1
%Fall 2018
%University of Notre Dame
%%%
clc
clear all
close all

A = exprnd(5,20,1);
mean_A = mean(A);
%Add code below
lambda_MLE = 
%Add code above
for alpha = 1:1:40
    beta = 100;
    n = length(A);
    %Add code below
    lambda_MAP = 
    %Add code above
    mse(alpha) = mean((0.2 - lambda_MAP).^2);
end
figure1 = figure;
axes1 = axes('Parent',figure1);
p1 = plot(mse,'k')
xlabel('\alpha','FontSize',15)
ylabel('MSE','FontSize',15)
set(p1,'LineWidth',2)
set(axes1,'FontSize',15,'FontWeight','bold');
saveas(figure1,'Solution_6b.png')

File: Homework/HW2/Assignment-2_Final/Python/Hm2-P6b.py
'''
Statistical Computing for Scientists and Engineers
Homework 2
Fall 2018
University of Notre Dame
'''
import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import expon
from sklearn.metrics import mean_squared_error
x = np.linspace(1,41,40)
A = np.random.exponential(scale=5,size=20)
##############################################################
# INSERT CODE BELOW
##############################################################

lambda_MLE = 
##############################################################
# INSERT CODE ABOVE
##############################################################

MAP = []
MAP_est= []
for alpha in range (1,41):
    beta = 100
    n = len(A)
##############################################################
# INSERT CODE BELOW
##############################################################
    lambda_MAP = 
##############################################################
# INSERT CODE ABOVE
##############################################################
    mse = ((0.2 - lambda_MAP) ** 2).mean(axis=None)
    MAP.append(mse)
MAP_lambda = np.array(MAP)
plt.xlabel(r'$\alpha$', fontsize = 16)
plt.ylabel('MSE', fontsize = 16)
plt.plot(x,MAP_lambda, marker='o', ms = 10, color='k')
plt.savefig('Solution-6B.png')
plt.show()


